tests/application_orchestrator_spec.rb

```diff
@@ -1,103 +1,117 @@
+      .exit_status
+      .should be_zero
+    file("#{MODULE_PATH}pasture_app/manifests/app.pp")
+      .content
+      .should match /define\s+pasture_app::app\s+\(.*?\$db_user,.*?\$db_password,.*?\$db_host,.*?/m
+    file("#{MODULE_PATH}pasture_app/manifests/app.pp")
+      .content
+      .should match /Pasture_App::App\s+consumes\s+Sql\s+{.*?db_user\s+=>\s+\$user,/m
   end
 end
 
-describe "Task 8:" do
-  it "Define a lamp::webapp component" do
-    # Need a better test!
-    file("#{MODULE_PATH}lamp/manifests/webapp.pp")
+describe 'Task 5:', host: :localhost do
+  it 'Create the pasture_app application class' do
+    file("#{MODULE_PATH}pasture_app/manifests/init.pp")
       .should be_file
+    command("puppet parser validate --app_management #{MODULE_PATH}pasture_app/manifests/init.pp")
+      .exit_status
+      .should be_zero
+    file("#{MODULE_PATH}pasture_app/manifests/init.pp")
+      .content
+      .should match /application\s+pasture_app\s+\(.*?\$db_user,.*?\$db_password,.*?/m
+    file("#{MODULE_PATH}pasture_app/manifests/init.pp")
+      .content
+      .should match /pasture_app::db\s+{\s+\$name:.*?export\s+=>\s+Sql\[\$name\],/m
+    file("#{MODULE_PATH}pasture_app/manifests/init.pp")
+      .content
+      .should match /pasture_app::app\s+{\s+\$name:.*?consume\s+=>\s+Sql\[\$name\],/m
   end
 end
 
-describe "Task 9:" do
-  it "Define the lamp application" do
-    # Need a better test!
-    file("#{MODULE_PATH}lamp/manifests/init.pp")
-      .should be_file
+describe 'Task 6:', host: :localhost do
+  it 'Remove application-related classes from role classes' do
+    file("#{MODULE_PATH}role/manifests/pasture_app.pp")
+      .content
+      .should match /class\s+role::pasture_app\s+{.*?include\s+profile::base::motd.*?include\s+profile::pasture::dev_users.*?}/m
+    file("#{MODULE_PATH}role/manifests/pasture_db.pp")
+      .content
+      .should match /class\s+role::pasture_db\s+{.*?include\s+profile::base::motd.*?include\s+profile::pasture::dev_users.*?}/m
   end
 end
 
-describe "Task 10:" do
-  it "Declare an application instance in your site.pp manifest" do
-    # Need a better test!
+describe 'Task 7:', host: :localhost do
+  it 'Add the application classification to site.pp' do
     file("#{PROD_PATH}manifests/site.pp")
       .content
-      .should match /Node\[\s*['"]database\.learning\.puppetlabs\.vm['"],?\s*\]\s*=>\s*Lamp::Mysql\[\s*['"]app1['"],?\s*\],/
+      .should match /site\s+{.*?pasture_app\s+{\s+(['"])pasture_01\1:.*?Node\[(['"])pasture-app-large\.puppet\.vm\1\]\s+=>\s+Pasture_app::App\[(['"])pasture_01\1\],/m
+    command("puppet parser validate --app_management #{PROD_PATH}manifests/site.pp")
+      .exit_status
+      .should be_zero
+    file('/root/.bash_history')
+      .content
+      .should match /puppet\s+job\s+plan\s+\-\-application\s+Pasture_app/
+    #command('/opt/puppetlabs/bin/puppet-job plan --application Pasture_app')
```

Comment:
@greglarkin Makes sense. I'm just going to merge this PR and create a new ticket to clean this up.